docs/tutorial/typesfuns.rst

```diff
@@ -1182,3 +1183,39 @@ matching ``let`` and lambda bindings. It will *only* work if:
 - The type of the result is "known". i.e. the type of the expression
   can be determined *without* type checking the ``case``-expression
   itself.
+
+Totality
+========
```

Comment:
Is it possible that the examples cover each type of total function. That is, provision of the 'safe head' and why it is total.